src/Components/Server/src/Builder/ServerRazorComponentsEndpointConventionBuilderExtensions.cs

```diff
@@ -17,7 +20,38 @@ public static class ServerRazorComponentsEndpointConventionBuilderExtensions
     /// <returns>The <see cref="RazorComponentsEndpointConventionBuilder"/>.</returns>
     public static RazorComponentsEndpointConventionBuilder AddInteractiveServerRenderMode(this RazorComponentsEndpointConventionBuilder builder)
     {
-        ComponentEndpointConventionBuilderHelper.AddRenderMode(builder, new InternalServerRenderMode());
+        return AddInteractiveServerRenderMode(builder, null);
+    }
+
+    /// <summary>
+    ///Maps the Blazor <see cref="Hub" /> to the default path.
+    /// </summary>
+    /// <param name="builder">The <see cref="RazorComponentsEndpointConventionBuilder"/>.</param>
+    /// <param name="callback">A callback to configure server endpoint options.</param>
+    /// <returns>The <see cref="ComponentEndpointConventionBuilder"/>.</returns>
+    public static RazorComponentsEndpointConventionBuilder AddInteractiveServerRenderMode(
+        this RazorComponentsEndpointConventionBuilder builder,
+        Action<ServerComponentsEndpointOptions>? callback = null)
+    {
+        var options = new ServerComponentsEndpointOptions();
+        callback?.Invoke(options);
+
+        ComponentEndpointConventionBuilderHelper.AddRenderMode(builder, new InternalServerRenderMode(options));
+
+        if ((options.EnableWebSocketCompression || options.ConfigureConnectionOptions is not null) && options.ContentSecurityFrameAncestorPolicy != null)
+        {
+            builder.AddEndpointFilter(new RequireCspFilter(options.ContentSecurityFrameAncestorPolicy));
```

Comment:
Do you think it might be easier to understand if we just apply this CSP to all Razor Component endpoints, independently of what render modes are registered?

I'm just thinking it will be very confusing for people if they set up a site containing page A that works in an iframe, and then later try to enable the InteractiveServer rendermode for page B, and for some reason page A in their site no longer works despite not changing any code related to page A. Is there some advantage to allowing framing conditionally based on configured rendermodes?

Reply:
I'm a bit worried about precluding the apps from being rendered inside an iframe by default, so I want to be more conservative. There's no reason for us to block the app from being embedded inside an iframe just because it uses wasm.

Reply:
We concluded that it's ok to just do this when we add server components as we believe that this will work in most scenarios (hosting ourselves inside an iframe) and hosting from other origins is much less common.